src/bitcoin/d/mod.rs

```diff
@@ -98,15 +95,7 @@ impl std::fmt::Display for BitcoindError {
                 f,
                 "Bitcoind server replied without enough responses to our batched request"
             ),
-            BitcoindError::WalletCreation(s) => write!(f, "Error creating watchonly wallet: {}", s),
-            BitcoindError::DescriptorImport(s) => write!(
-                f,
-                "Error importing descriptor. Response from bitcoind: '{}'",
-                s
-            ),
-            BitcoindError::WalletLoading(s) => {
-                write!(f, "Error when loading watchonly wallet: '{}'.", s)
-            }
+            BitcoindError::Wallet(_, e) => e.fmt(f),
```

Comment:
We should not ignore the path of the watchonly wallet file. It's always useful if there is an error with it.

```suggestion
            BitcoindError::Wallet(path, e) => write!(f, "Watchonly wallet (path: {}) error: {}", e),
```